pkg/kademlia/peer_discovery.go

```diff
@@ -96,11 +96,6 @@ func (lookup *peerDiscovery) Run(ctx context.Context) (_ []*pb.Node, err error)
 						)
 					}
 				} else {
-					if next.Id == lookup.target {
-						lookup.cond.L.Lock()
-						allDone = true
-						lookup.cond.L.Unlock()
-					}
```

Comment:
Is this above needed here or does it fix something?

Reply:
It was part of some testing code that I tried to make Jenkins run better, but it got accidentally committed.